amm/contracts/stable_pool/lib.rs

```diff
@@ -114,8 +114,17 @@ pub mod stable_pool {
     }
 
     #[ink(event)]
-    pub struct AmpCoefChanged {
-        pub new_amp_coef: u128,
+    pub struct AmpCoefChange {
+        pub initial_amp_coef: u128,
+        pub future_amp_coef: u128,
```

Comment:
How about adding also a getter like `future_amp_coef(&self) -> Option<(u128, u64)>` returning `None` if `current_time >= self.stop_amp_time`, and `Some((future_amp_coef, future_time))` otherwise? This would make some things a little bit easier in the indexer.